src/bundles/files/actions.js

```diff
@@ -22,57 +62,88 @@ const realMfsPath = (path) => {
   return path
 }
 
-const stat = async (ipfs, hashOrPath) => {
+/**
+ * @typedef {Object} Stat
+ * @property {string} path
+ * @property {'file'|'directory'|'unknown'} type
+ * @property {CID} cid
+ * @property {number} size
+ *
+ * @param {IPFSService} ipfs
+ * @param {string|CID} cidOrPath
+ * @returns {Promise<Stat>}
+ */
+const stat = async (ipfs, cidOrPath) => {
+  const hashOrPath = cidOrPath.toString()
   const path = hashOrPath.startsWith('/')
     ? hashOrPath
     : `/ipfs/${hashOrPath}`
 
   try {
     const stats = await ipfs.files.stat(path)
-    return stats
+    return { path, ...stats }
   } catch (e) {
     // Discard error and mark DAG as 'unknown' to unblock listing other pins.
     // Clicking on 'unknown' entry will open it in Inspector.
     // No information is lost: if there is an error related
     // to specified hashOrPath user will read it in Inspector.
+    const [, , cid] = path.split('/')
     return {
       path: hashOrPath,
-      hash: hashOrPath,
-      type: 'unknown'
+      cid: new CID(cid),
+      type: 'unknown',
+      size: -1
```

Comment:
This causes following problem https://github.com/ipfs-shipyard/ipfs-webui/pull/1569/#pullrequestreview-462879642

Reply:
maybe not